src/util/query.js

```diff
@@ -58,11 +58,17 @@ function parseQuery (query: string): Dictionary<string> {
     return res
   }
 
+  const illegalNames = Object.getOwnPropertyNames(Object.getPrototypeOf(res))
+
   query.split('&').forEach(param => {
     const parts = param.replace(/\+/g, ' ').split('=')
     const key = decode(parts.shift())
```

Comment:
let's do an `if (Object.prototype.hasOwnProperty(key)) return` here instead

Reply:
I thought this need should be solved by developers when they new Router with RouteConfig which contains "parseQuery / stringifyQuery". as a common router, this should not be filtered;

Reply:
After more thoughts, I don't think we need to skip properties in the native parsing of queries. Users should be able to still provide their own implementation